fastestimator/pipeline.py

```diff
@@ -92,6 +93,17 @@ def __init__(self,
         self.ctx_output_keys = None
         self.ctx_loader = None
 
+    def _register_ds_ids(self, data: Dict[str, Union[DataSource, Scheduler[DataSource], Dict[str, DataSource]]]):
+        """Associate dataset of each mode with a `ds_id`.
+
+        Args:
+            data: A dictionary with mode as key, dataset as value.
+        """
+        for mode, dataset in data.items():
+            if not isinstance(dataset, dict):
+                data[mode] = {None: dataset}
```

Comment:
1. key checking
2. possibly different default value

Reply:
added key checking in pipeline._register_ds_ids